language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -176,10 +176,10 @@ public static CodeActionNodeType topLevelNodeInLine(LSContext context, TextDocum
             Location location = diagnostic.location();
             LineRange lineRange = location.lineRange();
             TextRange textRange = location.textRange();
-            int startLine = lineRange.startLine().line(); // LSP diagnostics range is 0 based
-            int startChar = textRange.startOffset();
-            int endLine = lineRange.endLine().line();
-            int endChar = textRange.endOffset();
+            int startLine = lineRange.startLine().line() - 1; // LSP diagnostics range is 0 based
+            int startChar = textRange.startOffset() - 1;
```

Comment:
Should use `lineRange.startLine().offset()` instead of `textRange.startOffset()`